src/nvim/option.c

```diff
@@ -5012,7 +5012,7 @@ static int optval_default(vimoption_T *p, char_u *varp)
     return TRUE;            /* hidden option is always at default */
   dvi = ((p->flags & P_VI_DEF) || p_cp) ? VI_DEFAULT : VIM_DEFAULT;
   if (p->flags & P_NUM)
-    return *(intptr_t *)varp == (intptr_t)p->def_val[dvi];
+    return *(long *)varp == (long)p->def_val[dvi];
```

Comment:
If I am not mistaking, what actually should be here is `*(long *)varp == (long)(intptr_t)p->def_val[dvi]`: you can’t just go and cast pointer to integer without running into a warning, but you can first cast it to `u?intptr_t` integer first and then to the needed integer next.